apps/mobile/src/hooks/useDelegateCleanup/utils.ts

```diff
@@ -0,0 +1,311 @@
+  success: boolean
+  error?: string
+  failedDelegates?: Address[]
+}
+
+/**
+ * Cleans up notifications for all delegates of a given owner
+ * This is a critical step that must succeed before proceeding with delegate removal
+ */
+export const cleanupDelegateNotifications = async (
+  ownerAddress: Address,
+  delegateAddresses: Address[],
+  cleanupNotificationsForDelegate: (ownerAddress: Address, delegateAddress: Address) => Promise<CleanupResult>,
+): Promise<NotificationCleanupResult> => {
+  if (!delegateAddresses || delegateAddresses.length === 0) {
+    return { success: true }
+  }
+
+  try {
+    const notificationCleanupResults = await Promise.allSettled(
+      delegateAddresses.map(async (delegateAddress) => {
+        try {
+          const result = await cleanupNotificationsForDelegate(ownerAddress, delegateAddress)
+          if (!result.success && result.error) {
+            throw new Error(`Notification cleanup failed for ${delegateAddress}: ${result.error.message}`)
+          }
+          return result
+        } catch (error) {
+          Logger.error(`Failed to cleanup notifications for delegate ${delegateAddress}`, error)
+          throw error
+        }
+      }),
+    )
+
+    // Check if any notification cleanup failed with blocking errors
+    const failedCleanups = notificationCleanupResults
+      .filter((result) => result.status === 'rejected')
+      .map((result) => result.reason)
+
+    if (failedCleanups.length > 0) {
+      const failedDelegates = delegateAddresses.filter(
+        (_, index) => notificationCleanupResults[index].status === 'rejected',
+      )
+
+      const errorMsg = `Cannot delete private key: ${failedCleanups.join(', ')}. Please check your internet connection and try again.`
+
+      return {
+        success: false,
+        error: errorMsg,
+        failedDelegates,
+      }
+    }
+
+    return { success: true }
+  } catch (error) {
+    Logger.error('Delegate notification cleanup failed', error)
+    const errorMsg = error instanceof Error ? error.message : String(error)
+    return {
+      success: false,
+      error: errorMsg,
+      failedDelegates: delegateAddresses,
+    }
+  }
+}
+
+/**
+ * Removes delegates from the backend transaction service across all chains
+ * Uses rate limiting and retry logic to handle 429 errors and improve reliability
+ */
+export const removeDelegatesFromBackend = async (
+  ownerAddress: Address,
+  delegateAddresses: Address[],
+  ownerWallet: Wallet,
+  allChains: Chain[],
+  deleteDelegate: (params: DelegatesDeleteDelegateV2ApiArg) => Promise<unknown>,
+): Promise<DelegateRemovalResult> => {
+  if (!delegateAddresses || delegateAddresses.length === 0) {
+    return { success: true }
+  }
+
+  const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
+
+  // Retry function with exponential backoff
+  const retryWithBackoff = async <T>(fn: () => Promise<T>, maxRetries = 3, baseDelay = 1000): Promise<T> => {
```

Comment:
Can these be moved outside of the function maybe into their own util in case we want to reuse them?